lib/ws-channel.js

```diff
@@ -126,10 +138,13 @@ WebsocketChannel.prototype._detach = function() {
 
 
 WebsocketChannel.prototype.close = function(callback) {
+  this._debug('close');
+
   if (this._websocket) {
     this._unscheduleSendAck();
     this._flushSendQueue();
     this._closeWebsocket(this._websocket);
+    this._websocket = null;
```

Comment:
This isn't right. The normal flow is `_closeWebsocket()` -> `_handleClose()` -> `detach()` -> remove listeners and nullify.

You could opt to call `_detach()` directly instead of waiting for the `close` event.
But I deliberately made it so that the `_websocket` property has the same lifecycle as the event listeners that we add to the WS object, and I'd like to keep it that way.